src/arrays.js

```diff
@@ -1,61 +1,75 @@
 const getNthElement = (index, array) => {
-  // your code here
+       if(index >= array.length) {
+      index = index - array.length;
+        return array[index] 
+    }
+
+    else {
+        return array[index]
+	}
 };
 
 const arrayToCSVString = array => {
-  // your code here
+  	return array.toString()
 };
 
 const csvStringToArray = string => {
-  // your code here
+  	return string.split(",")
 };
 
 const addToArray = (element, array) => {
-  // your code here
+  array.push(element)
 };
 
 const addToArray2 = (element, array) => {
-  // your code here
+  newArray = array.concat(element)
+ return newArray
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+  array.splice(index, 1)
 };
 
 const numbersToStrings = numbers => {
-  // your code here
+  return numbers.map(String)
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  return strings.map(string => string.toUpperCase());
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+ // this one had me stumped! I tried many different ways but couldn't figure it out, please help! From what I can see, I believe it is some combination of Split(), reverse() and then join(), maybe it needs to iterate over each?
 };
 
 const onlyEven = numbers => {
-  // your code here
+  return numbers.filter( x => x % 2 === 0);
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+ remArray = array.slice(0, index).concat(array.slice((index + 1)));
+  return remArray 
+
+//this one almost drove me mad :). I didn't think to slice it either side of the index then combine for so long. It seemed counterintuitive for there not to be a way for slice to start after the index and then loop round to before the index!
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+    const words = strings.filter(string => string.toLowerCase().startsWith("a") || string.toLowerCase().startsWith("e") || string.toLowerCase().startsWith("i") || string.toLowerCase().startsWith("o") || string.toLowerCase().startsWith("u"));
+	return words
 };
 
 const removeSpaces = string => {
-  // your code here
+  return string.split(" ").join("");
 };
 
 const sumNumbers = numbers => {
-  // your code here
+   const reducer = (accumulator, currentValue) => accumulator + currentValue;
+return numbers.reduce(reducer)
+//still not 100% with reduce, I understand what is dynamically happening but cant figure out how the syntax reflects that.
 };
 
 const sortByLastLetter = strings => {
-  // your code here
+  // sorry Miguel, I spent a while on this but ran out of time - I think this one would involve splitting the array and then comparing the last characters of each word using charCodeAt() before using Sort() on the results in some way?
```

Comment:
You could actually use your reverseWords function from before.
Once that function is properly written, you can:

const reversed = reverseWordsInArray(strings);
return reversed.sort(); (which is another array methods - have a look!)